sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ArrayBasedMapBuilder.scala

```diff
@@ -64,7 +64,7 @@ class ArrayBasedMapBuilder(keyType: DataType, valueType: DataType) extends Seria
       throw QueryExecutionErrors.nullAsMapKeyNotAllowedError()
     }
 
-    val keyNormalized = keyNormalizer(key)
+    val keyNormalized = if (SQLConf.get.disableMapKeyNormalization) key else keyNormalizer(key)
```

Comment:
can we check this config when creating `keyNormalizer`?

Reply:
Refactored `keyNormalizer` to match the config as well.